src/form-sections/amount.ts

```diff
@@ -0,0 +1,347 @@
+import {
+  LitElement,
+  html,
+  PropertyValues,
+  nothing,
+  css,
+  TemplateResult,
+} from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import '@internetarchive/donation-form-section';
+import {
+  DonationPaymentInfo,
+  DonationType,
+} from '@internetarchive/donation-form-data-models';
+
+import type { MonthlyPlan } from '../models/plan';
+import '../presentational/ia-button';
+import type { IauxButton } from '../presentational/ia-button';
+import '../presentational/mgc-update-status';
+
+@customElement('ia-mgc-edit-plan-amount')
+export class MGCEditPlanAmount extends LitElement {
+  @property({ type: Object }) plan?: MonthlyPlan;
+
+  // captures the changes made to the amount && calculates the total amount with fees
+  @property({ type: Object }) donationPaymentInfo?: DonationPaymentInfo;
+
+  @property({ type: Number }) newAmount: number = 0;
+
+  @property({ type: Boolean, reflect: true }) currentlyEditing: boolean = false;
+
+  @property({ type: Boolean }) coverFees: boolean = false;
+
+  @property({ type: String }) updateMessage: string = '';
+
+  @property({ type: String }) errorMessage: string = '';
+
+  @property({ type: String }) updateStatus: 'success' | 'fail' | '' = '';
+
+  @query('form') form!: HTMLFormElement;
+
+  updated(changed: PropertyValues) {
+    if (changed.has('plan') && this.plan) {
+      this.captureAmountChanges();
+    }
+    if (changed.has('coverFees')) {
+      this.captureAmountChanges();
+    }
+    if (changed.has('currentlyEditing') && this.currentlyEditing) {
+      this.form.focus();
+    }
+
+    if (changed.has('donationPaymentInfo') && !this.donationPaymentInfo) {
+      this.captureAmountChanges();
+    }
+  }
+
+  protected render() {
+    return html`
+      <section>
+        <donation-form-section badgemode="hidebadge" headline="Amount">
+          ${!this.currentlyEditing
+            ? html` <p class="current-amount">
+                  USD $${this.plan?.amount}
+                  <ia-mgc-update-status .status=${this.updateStatus}
+                    >${this.updateMessage}</ia-mgc-update-status
+                  >
+                </p>
+                <ia-button
+                  id="edit-amount"
+                  class="ia-button link"
+                  .clickHandler=${() => {
+                    this.currentlyEditing = true;
+                    this.clearStatusMessaging();
+                  }}
+                >
+                  Edit...
+                </ia-button>`
+            : nothing}
+          ${this.currentlyEditing ? this.editAmountForm : nothing}
+        </donation-form-section>
+      </section>
+    `;
+  }
+
+  /* External Function that is called by consumer to tell the form there has been data update */
+  async amountUpdated(status: 'success' | 'fail') {
```

Comment:
I might make `status` a `@property` that gets updated by the consumer since it's the more common way of sending data between components.

Reply:
i currently want the minimum amount of properties to manage for this component. if i do, then i'll need an extra empty state to clear all the things. i'll hold this until a better abstraction shows up.